manager/eris-mint/evm/snative.go

```diff
@@ -1,104 +1,286 @@
-		registeredNativeContracts[LeftPadWord256([]byte("rm_role"))] = rm_role
-	*/
+type SNativeContractDescription struct {
+	Comment   string
+	Name      string
+	functions map[FuncID]*SNativeFunctionDescription
 }
 
-//-----------------------------------------------------------------------------
-// snative are native contracts that can access and modify an account's permissions
-
-type SNativeFuncDescription struct {
+type SNativeFunctionDescription struct {
+	Comment  string
 	Name     string
-	NArgs    int
+	Args     []abi.Arg
+	Return   abi.Return
 	PermFlag ptypes.PermFlag
 	F        NativeContract
 }
 
-/* The solidity interface used to generate the abi function ids below
-contract Permissions {
-	function has_base(address addr, uint64 permFlag) constant returns (bool value) {}
-	function set_base(address addr, uint64 permFlag, bool value) constant returns (bool val) {}
-	function unset_base(address addr, uint64 permFlag) constant returns (uint64 pf) {}
-	function set_global(uint64 permFlag, bool value) constant returns (uint64 pf) {}
-	function has_role(address addr, string role) constant returns (bool val) {}
-	function add_role(address addr, string role) constant returns (bool added) {}
-	function rm_role(address addr, string role) constant returns (bool removed) {}
+func registerSNativeContracts() {
+	for _, contract := range SNativeContracts() {
+		registeredNativeContracts[contract.Address()] = contract.Dispatch
+	}
 }
-*/
 
-// function identifiers from the solidity abi
-var PermsMap = map[string]SNativeFuncDescription{
-	getFuncIdentifiersFromSignature("has_role(address,bytes32)"):    SNativeFuncDescription{"has_role", 2, ptypes.HasRole, has_role},
-	getFuncIdentifiersFromSignature("unset_base(address,uint64)"):   SNativeFuncDescription{"unset_base", 2, ptypes.UnsetBase, unset_base},
-	getFuncIdentifiersFromSignature("set_global(uint64,bool)"):      SNativeFuncDescription{"set_global", 2, ptypes.SetGlobal, set_global},
-	getFuncIdentifiersFromSignature("add_role(address,bytes32)"):    SNativeFuncDescription{"add_role", 2, ptypes.AddRole, add_role},
-	getFuncIdentifiersFromSignature("set_base(address,uint64,bool"): SNativeFuncDescription{"set_base", 3, ptypes.SetBase, set_base},
-	getFuncIdentifiersFromSignature("has_base(address,uint64)"):     SNativeFuncDescription{"has_base", 2, ptypes.HasBase, has_base},
-	getFuncIdentifiersFromSignature("rm_role(address,bytes32)"):     SNativeFuncDescription{"rm_role", 2, ptypes.RmRole, rm_role},
+// Returns a map of all SNative contracts defined indexed by name
+func SNativeContracts() map[string]*SNativeContractDescription {
+	permFlagType := abi.Uint64
+	roleType := abi.Bytes32
+	contracts := []*SNativeContractDescription{
+		NewSNativeContract(`
+		* Interface for managing Secure Native authorizations.
+		* @dev This interface describes the functions exposed by the SNative permissions layer in the Monax blockchain (ErisDB).
+		`,
+			"permissions_contract",
```

Comment:
permissions_contract or are we doing PermissionsContract? Either way I'm fine. Just clarifying.